src/pyk/cterm/symbolic.py

```diff
@@ -88,17 +90,22 @@ def execute(
     ) -> CTermExecute:
         _LOGGER.debug(f'Executing: {cterm}')
         kore = self.kast_to_kore(cterm.kast)
-        response = self._kore_client.execute(
-            kore,
-            max_depth=depth,
-            cut_point_rules=cut_point_rules,
-            terminal_rules=terminal_rules,
-            module_name=module_name,
-            log_successful_rewrites=True,
-            log_failed_rewrites=self._trace_rewrites,
-            log_successful_simplifications=self._trace_rewrites,
-            log_failed_simplifications=self._trace_rewrites,
-        )
+        try:
+            response = self._kore_client.execute(
+                kore,
+                max_depth=depth,
+                cut_point_rules=cut_point_rules,
+                terminal_rules=terminal_rules,
+                module_name=module_name,
+                log_successful_rewrites=True,
+                log_failed_rewrites=self._trace_rewrites,
+                log_successful_simplifications=self._trace_rewrites,
+                log_failed_simplifications=self._trace_rewrites,
+            )
+        except SmtSolverError as err:
+            kast = self.kore_to_kast(err.pattern)
+            pretty_pattern = PrettyPrinter(self._definition).print(kast)
+            raise RuntimeError(f'\nSMT solver error:\n{pretty_pattern}') from err
```

Comment:
You can extract a method:

```python3
def _smt_solver_error(self, err: SmtSolverError) -> RuntimeError:
    kast = self.kore_to_kast(err.pattern)
    pretty_pattern = PrettyPrinter(self._definition).print(kast)
    return RuntimeError(f'\nSMT solver error:\n{pretty_pattern}')
```

Then in all four places:

```suggestion
        except SmtSolverError as err:
            raise self._smt_solver_error(err) from err
```

Reply:
Also, consider removing the leading newline:

```python3
RuntimeError(f'SMT solver error:\n{pretty_pattern}')
```